File: lib/messages/race.dart
class Race {
  final String what = 'race';
  final String mode;

  String title;
  int tracks;
  int rounds;

  Race({
    required this.mode,
    this.title = '',
    this.tracks = 0,
    this.rounds = 0,
  });

  Race.fromJson(Map<String, dynamic> json)
      : mode = json['mode'] as String,
        title = json['title'] as String,
        tracks = json['tracks'] as int,
        rounds = json['rounds'] as int;

  Map<String, dynamic> toJson() => {
        'what': what,
        'mode': mode,
        'title': title,
        'tracks': tracks,
        'rounds': rounds,
      };
}

File: soft/golang/slotman/services/type/slotman/race.go
package slotman

import (
	"slotman/utils/simple"
	"time"
)

type RaceState string

const (
	RaceStateIdle          RaceState = "state.idle"
	RaceStateRaceWaiting   RaceState = "state.race.waiting"
	RaceStateRaceStarting  RaceState = "state.race.start"
	RaceStateRaceRunning   RaceState = "state.race.running"
	RaceStateRaceSuspended RaceState = "state.race.suspended"
	RaceStateRaceFinished  RaceState = "state.race.finished"
)

type Race struct {
	What string `json:"what,omitempty"`
	Mode string `json:"mode,omitempty"`

	Title string `json:"title"`

	Tracks int `json:"tracks"`
	Rounds int `json:"rounds"`
}

type RaceInfo struct {
	What string `json:"what,omitempty"`
	Mode string `json:"mode,omitempty"`

	Track    int `json:"track"`
	Rounds   int `json:"rounds"`
	Position int `json:"position"`

	ActRound float64 `json:"actRound"`
	TopRound float64 `json:"topRound"`

	ActSpeed float64 `json:"actSpeed"`
	TopSpeed float64 `json:"topSpeed"`

	PilotUuid simple.UUIDHex `json:"pilotUuid"`

	LastRoundTime time.Time `json:"-"`
}

type TrackState int

const (
	TrackStateInactive TrackState = 0
	TrackStateActive   TrackState = 1
	TrackStateReady    TrackState = 2
	TrackStateFinished TrackState = 3
)
